### DISCUSSION TITLE
A question about extracting viral forces from cp2k data

### DISCUSSION THREAD

**Discussion by TerryHangLyu** [deepmodeling/deepmd-kit#1538] [Q&A]
Dear All,
   Recently, I got a series of MD trajectories from CP2K, and I used dpdata to convert the MD trajectories to npy for deepmd-kit. But I found that though I asked cp2k to print the stress tensor to the .log and save the stress tensor to the .stress file, dpdata could not read the stress tensor and make a virial.npy which can be generated when the AIMD code is VASP.
   And I check the example files here https://github.com/deepmodeling/dpdata/tree/master/tests/cp2k/aimd/deepmd/set.000,  it also doesn't contain the virial.npy. 
   My question is how to generate the virial.npy from the cp2k data.

Best wishes,
Hang LYU

**Comment by njzjz** [comment]
@robinzyb may answer the question..
